fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/ExecuteState.java

```diff
@@ -35,27 +35,35 @@
  */
 @API(API.Status.MAINTAINED)
 public class ExecuteState {
+
     /**
-     * An empty execute state with no record scan limit.
-     * @deprecated in favor of NO_LIMITS when the byte scan limit was added
+     * An execute state with no scan limits.
      */
-    @API(API.Status.DEPRECATED)
-    @Deprecated
-    public static final ExecuteState NO_SCANNED_RECORDS_LIMIT = new ExecuteState();
+    public static final ExecuteState NO_LIMITS = new ExecuteState(RecordScanLimiter.Untracked.INSTANCE, ByteScanLimiter.Untracked.INSTANCE);
 
     /**
-     * An empty execute state with no record scan limit.
+     * An empty execute state with no scan limits.
+     * @deprecated in favor of NO_LIMITS when the byte scan limit was added
      */
-    public static final ExecuteState NO_LIMITS = new ExecuteState();
+    @API(API.Status.DEPRECATED)
+    @Deprecated
+    public static final ExecuteState NO_SCANNED_RECORDS_LIMIT = NO_LIMITS;
 
-    @Nullable
+    @Nonnull
     private final RecordScanLimiter recordScanLimiter;
-    @Nullable
+    @Nonnull
     private final ByteScanLimiter byteScanLimiter;
 
+    /**
+     * Creates an execute state with a supplied set of resource limiters.
+     * @param recordScanLimiter a record scan limiter or {@code null} to indicate an unlimited
+     *     number of records may be scanned
+     * @param byteScanLimiter a byte scan limiter or {@code null} to indicate an unlimited
+     *     number of bytes may be scanned
+     */
     public ExecuteState(@Nullable RecordScanLimiter recordScanLimiter, @Nullable ByteScanLimiter byteScanLimiter) {
```

Comment:
We should deprecate this method and create a new one with `Nonnull` arguments.

Reply:
Oh no! Perhaps we should just change the API then? Or at least create an issue to do so in Record Layer 3?